src/ecephys_etl/modules/vbn_create_stimulus_table/create_stim_table.py

```diff
@@ -0,0 +1,551 @@
+    frame_timestamps = get_vsyncs(sync_dataset)
+    frame_timestamps = frame_timestamps[frame_offset:frame_offset + num_frames]
+
+    check_behavior_and_replay_pkl_match(
+        replay_pkl=replay_pkl, behavior_stim_table=behavior_stim_table
+    )
+
+    # If replay pkl data and behavior pkl data match, use the existing
+    # behavior stim table but adjust the times/frames
+    stim_table = behavior_stim_table.copy(deep=True)
+    stim_table['stimulus_block'] = block_offset
+    stim_table['start_time'] = frame_timestamps[stim_table['start_frame']]
+    stim_table['stop_time'] = frame_timestamps[stim_table['end_frame']]
+    stim_table['duration'] = stim_table['stop_time'] - stim_table['start_time']
+    stim_table['start_frame'] = stim_table['start_frame'] + frame_offset
+    stim_table['end_frame'] = stim_table['end_frame'] + frame_offset
+    stim_table['active'] = False
+
+    column_dtypes = {
+        "duration": float,
+        "end_frame": int,
+        "start_frame": int,
+        "start_time": float,
+        "stop_time": float,
+        "stimulus_block": int,
+        "active": bool
+    }
+
+    return stim_table.astype(column_dtypes)
+
+
+def generate_mapping_stim_table(
+    mapping_pkl: CamStimOnePickleStimFile,
+    sync_dataset: Dataset,
+    frame_offset: int = 0
+) -> pd.DataFrame:
+
+    def seconds_to_frames(
+        seconds: List[Union[int, float]]
+    ) -> List[Union[int, float]]:
+        offset_times = np.array(seconds) + mapping_pkl.pre_blank_sec
+        return offset_times * mapping_pkl.frames_per_second
+
+    stim_tabler = partial(
+        build_stimuluswise_table, seconds_to_frames=seconds_to_frames
+    )
+    stim_df = create_stim_table(
+        mapping_pkl.stimuli, stim_tabler, make_spontaneous_activity_tables
+    )
+
+    frame_timestamps = get_vsyncs(sync_dataset)
+
+    stim_df = stim_df.rename(
+        columns={
+            "Start": "start_frame",
+            "End": "end_frame",
+            "TF": "temporal_frequency",
+            "SF": "spatial_frequency",
+            "Ori": "orientation",
+            "Contrast": "contrast",
+            "Pos_x": "position_x",
+            "Pos_y": "position_y",
+            "Color": "color"
+        }
+    )
+
+    # Fill in table row(s) for 'spontaneous' stimuli
+    # Characterized by NaN for "stimulus_name" and "stimulus_block"
+    spont_rows = (
+        stim_df["stimulus_name"].isnull() & stim_df["stimulus_block"].isnull()
+    )
+    stim_df.loc[spont_rows, "stimulus_name"] = 'spontaneous'
+    stim_df.loc[spont_rows, "stimulus_block"] = 2
+
+    # Update "stimulus_block" values for gabor and flash stimuli
+    # Gabor "stimulus_block" should take on value of 1
+    gabor_rows = stim_df["stimulus_name"].str.contains("gabor")
+    stim_df.loc[gabor_rows, "stimulus_block"] = 1
+    # Flash "stimulus_block" should take on value of 3
+    flash_rows = stim_df["stimulus_name"].str.contains("flash")
+    stim_df.loc[flash_rows, "stimulus_block"] = 3
```

Comment:
@corbennett Hi Corbett,

    Just wanted to double check that these stimulus_block values are what were intended. Because a mapping session looks like:

    'spontaneous'
    'gabor stimuli'
    'flashes'
    'spontaneous'

The block_offset parameter in the original un-edited version of the function doesn't really make sense since the raw mapping table gives gabor stimuli a stimulus_block = 1 and flash stimuli a stimulus_block = 2.

    Also wanted to check that searching for gabor and flash is a reliable way to find gabor and flash presentations? I'm pretty confident, but wanted to double check with you since you know the pkl files structure for the mapping pkl a lot better than me.